src/m_menu.c

```diff
@@ -3189,10 +3189,27 @@ static const char *hudtype_strings[] = {
     "Crispy", "No Bars", "Boom"
 };
 
-// This matches the WOOFHUD documentation.
-static const char *hudmode_strings[] = {
-    "Minimal", "Compact", "Distributed"
-};
+static const char **M_GetHUDModeStrings(void)
+{
+    const char *crispy_strings[] = {"Off", "Original", "Widescreen"};
+    const char *boom_strings[] = {"Minimal", "Compact", "Distributed"};
+    const char **current_strings = hud_type ? boom_strings : crispy_strings;
+    static const char **hudmode_strings = NULL;
+
+    if (hudmode_strings)
+    {
+        array_free(hudmode_strings);
+    }
```

Comment:
I think there is no need in dynamic array, we can just return static strings:
```diff
diff --git a/src/m_menu.c b/src/m_menu.c
index 05483a9..00605f3 100644
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -3191,22 +3191,9 @@ static const char *hudtype_strings[] = {
 
 static const char **M_GetHUDModeStrings(void)
 {
-    const char *crispy_strings[] = {"Off", "Original", "Widescreen"};
-    const char *boom_strings[] = {"Minimal", "Compact", "Distributed"};
-    const char **current_strings = hud_type ? boom_strings : crispy_strings;
-    static const char **hudmode_strings = NULL;
-
-    if (hudmode_strings)
-    {
-        array_free(hudmode_strings);
-    }
-
-    for (int i = 0; i < 3; i++)
-    {
-        array_push(hudmode_strings, current_strings[i]);
-    }
-
-    return hudmode_strings;
+    static const char *crispy_strings[] = {"Off", "Original", "Widescreen"};
+    static const char *boom_strings[] = {"Minimal", "Compact", "Distributed"};
+    return hud_type ? boom_strings : crispy_strings;
 }
 
 static void M_UpdateHUDModeStrings(void);
```

Reply:
Yes, that's much better. Thank you. Fixed here: https://github.com/fabiangreffrath/woof/pull/1423/commits/72ff5d3418862f75ae7ef7ef9e7e9350cb35467b